docs/api/web-contents.md

```diff
@@ -265,6 +265,31 @@ this purpose.
 
 Calling `event.preventDefault()` will prevent the navigation.
 
+#### Event: 'will-frame-navigate'
+
+Returns:
+
+* `event` Event
+* `url` String
+* `isInPlace` Boolean
+* `isMainFrame` Boolean
+* `frameProcessId` Integer
+* `frameRoutingId` Integer
+
+Emitted when a user or the page wants to start navigation in any frame. It can happen when
+the `window.location` object is changed or a user clicks a link in the page.
+
+Unlike `will-navigate`, `will-frame-navigate` is fired when the main frame or any of its subframes attempts to navigate. When the navigation event comes from the main frame, `isMainFrame` will be `true`.
+
+This event will not emit when the navigation is started programmatically with
+APIs like `webContents.loadURL` and `webContents.back`.
+
+It is also not emitted for in-page navigations, such as clicking anchor links
+or updating the `window.location.hash`. Use `did-navigate-in-page` event for
+this purpose. This means that `isInPlace` will always be `false`.
```

Comment:
```suggestion
this purpose. This means that `isInPlace` will always be `false`.
```

Why do we expose this flag if it's always false?

Reply:
I left that parameter there because I'm using [EmitNavigationEvent](https://github.com/electron/electron/blob/1cd07c565a87ced6a923ade76eda380367e5ce1f/shell/browser/api/electron_api_web_contents.cc#L1741) which is used by several navigation-related events on `webContents`, and always emits these 6 positional arguments. I can modify `EmitNavigationEvent` to take in a parameter to decide whether to emit the `isInPlace` event. That feels a little weird, but it should work.

Ideally we'd emit all these arguments in a single dictionary and just not document the irrelevant ones, but of course that would be a pretty significant breaking change.